src/options.cpp

```diff
@@ -256,10 +257,27 @@ void long_usage(char const *arg0, bool verbose)
 
 } // anonymous namespace
 
+static bool compare_prefix(std::string const &str,
+                           std::string const &prefix) noexcept
+{
+    return std::strncmp(str.c_str(), prefix.c_str(), prefix.size()) == 0;
+}
+
 std::string database_options_t::conninfo() const
 {
+    if (compare_prefix(db, "postgresql://") ||
+        compare_prefix(db, "postgres://")) {
+        return db;
+    }
+
     std::string out{"fallback_application_name='osm2pgsql'"};
```

Comment:
Should this be above the block for `postgresql://` handling? We still want to set the fallback application name

Reply:
No. In that case we can't set the application name if I understand the postgresql docs correctly. These are two totally different ways of specifying the connection string and they can't be mixed.